code/modules/antagonists/changeling/powers/headcrab.dm

```diff
@@ -19,13 +19,13 @@
 			if(other_ling?.isabsorbing)
 				to_chat(user, "<span class='warning'>Our last resort is being disrupted by another changeling!</span>")
 				return
-	var/turf/T = user.loc
-	if(!isopenturf(T))
-		to_chat(user, "<span class='warning'>You need an open space to become a headslug!</span>")
-		return
 	if(alert("Are we sure we wish to kill ourself and create a headslug?",,"Yes", "No") == "No")
 		return
 	..()
+	var/turf/T = user.loc
+	if(!isopenturf(T) || !T || !is_changeling(user))
```

Comment:
So someone could actually open the alert wait to die and then just accept? I guess thats why you check for is_changeling here

Reply:
Yes, that was infact possible

Reply:
Also maybe check for !T first since its a cheaper check than istype inside the isopenturf macro

Reply:
Sure

Reply:
Technically the istype check will also return null if the var is null but still might be cheaper (a little) to just check !T in the beginning in its current position even if T=null it won't ever have the chance to fire since that would just end at the typcheck